File: Python/programmers_완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    dic = {}
    for person in participant:
        if person in dic:
            dic[person] = dic[person] + 1
        else:
            dic[person] = 1
    
    for person in completion:
        if person in dic:
            if dic[person] == 1:
                del dic[person]
            else:
                dic[person] = dic[person] - 1
    
    for key in dic:
        answer = key
    
    return answer

File: JS/programmers/programmers_완주하지 못한 선수.js
function solution(participant, completion) {
    var answer = '';
    const map = new Map();
    participant.forEach(person => {
        if (!map.has(person)) 
            map.set(person, 1);
        else
            map.set(person, map.get(person) + 1);
    })
    for(const person of completion) {
        if (map.get(person) === 1) {a
            map.delete(person)
        } else {
            map.set(person, map.get(person) - 1);
        }
    }
    console.log(map)
    for (const key of map.keys()) {
        return key;
    }
    
}